src/ui/viewmodels/MemoryViewerViewModel.hh

```diff
@@ -235,6 +239,8 @@ private:
     static std::unique_ptr<ra::ui::drawing::ISurface> s_pFontSurface;
     static std::unique_ptr<ra::ui::drawing::ISurface> s_pFontASCIISurface;
     static int s_nFont;
+    static std::list<ra::ByteAddress> s_listMemViewHistory;
+    static std::list<ra::ByteAddress>::iterator s_iterMemViewHistoryIndex;
```

Comment:
Suggest `s_iterMemViewHistoryIndex` just be an integer. Maintaining a iterator over a dynamically sized array can be difficult. Also, I think it would improve readability.

Reply:
I switch to vector and refactor the logic on the last commit. I also tried to switch to integer index instead of iterator but I found it way less readable than the iterator version.

There's probably a better way to do it so I post it that version here and let you judge which version should be implemented and/or what to improve : 

```
void MemoryViewerViewModel::AddToHistory(ra::ByteAddress nAddress)
{
    if (vHistory.at(nHistoryIndex) != nAddress)
    {
        vHistory.erase(std::next(vHistory.begin(), nHistoryIndex + 1), vHistory.end());
        if (vHistory.size() == MEMVIEW_HISTORY_MAX_SIZE)
            vHistory.erase(vHistory.begin());
        vHistory.push_back(nAddress);
        nHistoryIndex = vHistory.size() - 1;
    }
}

void MemoryViewerViewModel::MoveHistoryForward()
{
    int nIndex = std::distance(vHistory.begin(), std::next(vHistory.begin(), nHistoryIndex));
    if (vHistory.size() > (nIndex + 1))
        ++nHistoryIndex;
    SetAddress(vHistory.at(nHistoryIndex));
}

void MemoryViewerViewModel::MoveHistoryBackward()
{
    if (vHistory.size() >= 1 and std::next(vHistory.begin(), nHistoryIndex) != vHistory.begin())
        --nHistoryIndex;
    SetAddress(vHistory.at(nHistoryIndex));
}
```

VS
```
void MemoryViewerViewModel::AddToHistory(ra::ByteAddress nAddress)
{
    if (*iterHistoryIndex != nAddress)
    {
        vHistory.erase(iterHistoryIndex + 1, vHistory.end());
        if (vHistory.size() == MEMVIEW_HISTORY_MAX_SIZE)
            vHistory.erase(vHistory.begin());
        vHistory.push_back(nAddress);
        iterHistoryIndex = vHistory.end() - 1;
    }
}

void MemoryViewerViewModel::MoveHistoryForward()
{
    int nIndex = std::distance(vHistory.begin(), iterHistoryIndex);
    if (vHistory.size() > (nIndex + 1))
        ++iterHistoryIndex;
    SetAddress(*iterHistoryIndex);
}

void MemoryViewerViewModel::MoveHistoryBackward()
{
    if (vHistory.size() >= 1 and iterHistoryIndex != vHistory.begin())
        --iterHistoryIndex;
    SetAddress(*iterHistoryIndex);
}
```